### PR TITLE
Feature/bokbok

### PR DESCRIPTION
**Pull request by AlexDunmow**

### CODE REVIEW

**Comment by AlexDunmow** [comment]
depends on https://github.com/perthgophers/puddle/pull/35

**Comment by AlexDunmow** [comment]
Only writes to #puddletest

File: responses/bokbok.go [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+		tx.CreateBucketIfNotExists([]byte("all"))
+		return nil
+	})
+
+	bkbk.prefixLen = prefixLen
+
+	return bkbk
+}
+
+// Processes Message builds or amends the Markov Chain for "all" and the individual user
```
Comment (nii236) [inline responses/bokbok.go:52]: `// ProcessMessage...`

File: responses/bokbok.go [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+	allch := bkbk.Chain("all")
+	allch.BuildFromString(cr.Text)
+
+	bkbk.SaveChains(ch, allch)
+	return nil
+}
+
+// MaybeRespond might respond, or it might not, for top kek
+func (bkbk *BokBok) MaybeRespond(cr *messagerouter.CommandRequest, w messagerouter.ResponseWriter) error {
+	if cr.Message.Channel == "C33C4MJSH" && bkbk.YesNo() {
```
Comment (nii236) [inline responses/bokbok.go]: There shouldn't be a need to hardcode the channel here. You  pass in `mr` into responses.Init() which has `mr.SpamChannel`

**Review by nii236 — APPROVED** [review]

**Comment by nii236** [comment]
LGTM

File: responses/bokbok.go (hunk 2 above)
Comment (AlexDunmow) [reply responses/bokbok.go]: Actually, spamchannel is unexported.

Should mr.channel & mr.spamchannel be changed to mr.Channel & mr.SpamChannel respectively?

Comment (nii236) [reply responses/bokbok.go]: Isn't there a package global in the `responses` package?

**Pull request merged**